src/coreclr/jit/optimizer.cpp

```diff
@@ -7645,9 +7665,6 @@ void Compiler::optHoistCandidate(GenTree* tree, BasicBlock* treeBb, unsigned lnu
         return;
     }
 
-    // Create a loop pre-header in which to put the hoisted code.
-    fgCreateLoopPreHeader(lnum);
```

Comment:
You can't move this down because the next check depends on the pre-header already being created.